build.sbt

```diff
@@ -297,7 +297,7 @@ val scalaCheckVshort      = scalaCheckV.take(4).replace(".", "-")
 val scalaTestPlusV        =
   "3.2.18.0" // has to match scalatest and scalacheck versions, see https://github.com/scalatest/scalatestplus-scalacheck/releases
 // note: Logback 1.3 requires Slf4j 2.x, but Flink has Slf4j 1.7 on its classpath
-val logbackV                = "1.2.13"
+val logbackV                = "1.5.12"
```

Comment:
Are you sure you want to change it? Flink 1.20 provides `org.slf4j:slf4j-api:jar:1.7.36`, which is incompatible with this.

Reply:
No, you're right, I added this just to check if there are any incompatibilities like with 1.19 (there doesn't seem to be) and to check if tests pass besides this.

With previous versions of logback and slf4j when launching a Flink minicluster tests failed with:
`java.lang.NullPointerException: null
	at java.base/java.util.HashMap.putMapEntries(HashMap.java:497)
	at java.base/java.util.HashMap.putAll(HashMap.java:781)
	at java.base/java.util.Collections$SynchronizedMap.putAll(Collections.java:2604)
	at ch.qos.logback.classic.util.LogbackMDCAdapter.setContextMap(LogbackMDCAdapter.java:197)
	at org.slf4j.MDC.setContextMap(MDC.java:264)
	at org.apache.flink.util.MdcUtils.lambda$withContext$0(MdcUtils.java:48)
	at org.apache.flink.runtime.rpc.pekko.PekkoRpcActor.handleControlMessage(PekkoRpcActor.java:208)
`

After bumping the versions, the tests run ok. I wonder why our e2e tests dont blow up if we add an incompatible dependency like this :/ I'll check this

Anyways, there is a ticket for this problem: https://issues.apache.org/jira/browse/FLINK-36227 that is yet to be addressed.

I'll revert this and slf4j version bumps, see if there's a workaround and if not I'll monitor the Flink jira ticket. Thanks!